serial_scripts/solution/topology/mini_deployment/vsrx_config/vrp32_config.txt
#junos-config - vEPC vRP nodes

system {
    root-authentication {
        encrypted-password "$6$sg7yK0wP$XKPfVjffdOp6g38R1UXui4Lxx5QZHARNdNfgkpXofFr7hT5R2sjUyFrdo5AitBujdpGvAOy2ziJG4oS8jOoiu1";
    }
    login {
        user admin {
            uid 2000;
            class super-user;
            authentication {
                encrypted-password "$6$sg7yK0wP$XKPfVjffdOp6g38R1UXui4Lxx5QZHARNdNfgkpXofFr7hT5R2sjUyFrdo5AitBujdpGvAOy2ziJG4oS8jOoiu1";
            }
        }
    }
    host-name vrp-32 
    services {
        ssh {
            root-login allow;
        }
        web-management {
            http {
                interface fxp0.0;
            }
        }
    }
    syslog {
        file messages {
            any any;
        }
    }
}

security {
    forwarding-options {
        family {
            inet6 {
                mode packet-based;
            }
            mpls {
                mode packet-based;
            }
            iso {
                mode packet-based;
            }
        }
    }
}

interfaces {
    fxp0 {
        unit 0 {
            description "vSRX OOB Management interface";
            family inet {
                address 192.168.25.32/24;
            }
        }
    }
    ge-0/0/0 {
        unit 0 {
            description "vEPC RP MGNT interface";
            family inet {
                dhcp;
            }
        }
    }
    ge-0/0/1 {
        unit 0 {
            description "vEPC RP BP interface";
            family inet {
                address 192.3.1.32/16;
            }
        }
    }
    ge-0/0/2 {
        unit 0 {
            description "vEPC RP MATE interface";
            family inet {
                address 127.4.1.32/16;
            }
        }
    }
    lo0 {
        unit 0 {
            description "vEPC RP BP - OSPF LoopBack";
            family inet {
                address 172.31.2.32/32;
            }
        }
    }
}

protocols {
    ospf {
        area 0.0.0.0 {
            interface ge-0/0/1.0 {
                priority 198;
                hello-interval 1;
                dead-interval 3;
            }
            interface lo0.0 {
                passive;
            }
        }
    }
}

routing-options {
    static {
        route 0.0.0.0/1 next-hop 192.168.25.1;
        route 128.0.0.0/1 next-hop 192.168.25.1; 
    }
    router-id 172.31.2.32;
}


serial_scripts/solution/topology/mini_deployment/vsrx_config/vsfo1_config.txt
#junos-config - vEPC VSFO Control Plane

system {
    root-authentication {
        encrypted-password "$6$sg7yK0wP$XKPfVjffdOp6g38R1UXui4Lxx5QZHARNdNfgkpXofFr7hT5R2sjUyFrdo5AitBujdpGvAOy2ziJG4oS8jOoiu1";
    }
    login {
        user admin {
            uid 2000;
            class super-user;
            authentication {
                encrypted-password "$6$sg7yK0wP$XKPfVjffdOp6g38R1UXui4Lxx5QZHARNdNfgkpXofFr7hT5R2sjUyFrdo5AitBujdpGvAOy2ziJG4oS8jOoiu1";
            }
        }
    }
    host-name vsfo_cp-1 
    services {
        ssh {
            root-login allow;
        }
        web-management {
            http {
                interface fxp0.0;
            }
        }
    }
    syslog {
        file messages {
            any any;
        }
    }
}

security {
    forwarding-options {
        family {
            inet6 {
                mode packet-based;
            }
            mpls {
                mode packet-based;
            }
            iso {
                mode packet-based;
            }
        }
    }
}

interfaces {
    fxp0 {
        unit 0 {
            description "vSRX OOB Management interface";
            family inet {
                address 192.168.25.101/24;
            }
        }
    }
    ge-0/0/0 {
        unit 0 {
            description "vEPC VSFO CP BP interface";
            family inet {
                address 192.3.1.101/16;
            }
        }
    }
    ge-0/0/1 {
        unit 0 {
            description "vEPC VSFO CP VFAB 1 interface";
            family inet {
                address 127.6.21.101/24;
            }
        }
    }
    ge-0/0/2 {
        unit 0 {
            description "vEPC VSFO CP VFAB 2 interface";
            family inet {
                address 127.6.22.101/24;
            }
        }
    }
    ge-0/0/3 {
        unit 0 {
            description "vEPC VSFO CP VFAB 3 interface";
            family inet {
                address 127.6.23.101/24;
            }
        }
    }
    ge-0/0/4 {
        unit 0 {
            description "vEPC VSFO CP VFAB 4 interface";
            family inet {
                address 127.6.24.101/24;
            }
        }
    }
    ge-0/0/5 {
        flexible-vlan-tagging;
        native-vlan-id 1;
        unit 0 {
            description "vEPC VSFO CP EXT 1 - ROOT interface";
            vlan-id 1;
            family inet {
                address 172.21.1.3/24;
            }
        }
        unit 1001 {
            description "vEPC VSFO CP EXT 1 - O&M, Sig_CN, Media_CN sub-interface";
            vlan-id 1001;
            family inet {
                address 11.10.1.3/29;
            }
        }
        unit 1002 {
            description "vEPC VSFO CP EXT 1 - O&M, Sig_CN, Media_CN sub-interface";
            vlan-id 1002;
            family inet {
                address 11.10.2.3/29;
            }
        }
        unit 1003 {
            description "vEPC VSFO CP EXT 1 - O&M, Sig_CN, Media_CN sub-interface";
            vlan-id 1003;
            family inet {
                address 11.10.3.3/29;
            }
        }
        unit 2000 {
            description "vEPC VSFO CP EXT 1 - shared radius + APN dedicated radius sub-interface";
            vlan-id 2000;
            family inet {
                address 11.20.0.3/29;
            }
        }
        unit 2001 {
            description "vEPC VSFO CP EXT 1 - shared radius + APN dedicated radius sub-interface";
            vlan-id 2001;
            family inet {
                address 11.20.1.3/29;
            }
        }
        unit 2002 {
            description "vEPC VSFO CP EXT 1 - shared radius + APN dedicated radius sub-interface";
            vlan-id 2002;
            family inet {
                address 11.20.2.3/29;
            }
        }
        unit 2003 {
            description "vEPC VSFO CP EXT 1 - shared radius + APN dedicated radius sub-interface";
            vlan-id 2003;
            family inet {
                address 11.20.3.3/29;
            }
        }
        unit 2004 {
            description "vEPC VSFO CP EXT 1 - shared radius + APN dedicated radius sub-interface";
            vlan-id 2004;
            family inet {
                address 11.20.4.3/29;
            }
        }
        unit 2005 {
            description "vEPC VSFO CP EXT 1 - shared radius + APN dedicated radius sub-interface";
            vlan-id 2005;
            family inet {
                address 11.20.5.3/29;
            }
        }
    }
    lo0 {
        unit 0 {
            description "vEPC VSFO CP BP - OSPF LoopBack";
            family inet {
                address 172.31.2.1/32;
            }
        }
        unit 1001 {
            description "vEPC VSFO CP Sig 1001 Loopback";
            family inet {
                address 11.10.1.255/32;
            }
        }
        unit 1002 {
            description "vEPC VSFO CP Sig 1002 Loopback";
            family inet {
                address 11.10.2.255/32;
            }
        }
        unit 1003 {
            description "vEPC VSFO CP Sig 1003 Loopback";
            family inet {
                address 11.10.3.255/32;
            }
        }
        unit 2000 {
            description "vEPC VSFO CP Radius 2000 Loopback";
            family inet {
                address 11.20.0.255/32;
            }
        }
        unit 2001 {
            description "vEPC VSFO CP Radius 2001 Loopback";
            family inet {
                address 11.20.1.255/32;
            }
        }
        unit 2002 {
            description "vEPC VSFO CP Radius 2002 Loopback";
            family inet {
                address 11.20.2.255/32;
            }
        }
        unit 2003 {
            description "vEPC VSFO CP Radius 2003 Loopback";
            family inet {
                address 11.20.3.255/32;
            }
        }
        unit 2004 {
            description "vEPC VSFO CP Radius 2004 Loopback";
            family inet {
                address 11.20.4.255/32;
            }
        }
        unit 2005 {
            description "vEPC VSFO CP Radius 2005 Loopback";
            family inet {
                address 11.20.5.255/32;
            }
        }
    }
}

policy-options {
    policy-statement REDISTRIBUTE_TO_BGP {
        term direct_to_bgp {
            from {
                protocol direct;
                route-filter 0.0.0.0/0 prefix-length-range /32-/32;
            }
            then accept;
        }
    }
}

routing-instances {
    sig_1001_vpn {
        instance-type virtual-router;
        description "Signaling VEPC VPN - 1001";
        interface ge-0/0/5.1001;
        interface lo0.1001;
        routing-options {
            autonomous-system 65001;
            static {
                route 1.0.0.1/32 {
                    next-hop 11.10.1.1;
                    metric 254;
                    bfd-liveness-detection {
                        description "BFD to vrouter primary IP."
                        minimum-interval 1000;
                        minimum-receive-interval 1000;
                        transmit-interval {
                            minimum-interval 1000;
                        }
                        multiplier 3;
                        session-mode multihop;
                     }
                }
            }
        }
        protocols {
            bgp {
                group sig_vsfo1ext1vlan1001 {
                    type external;
                    export REDISTRIBUTE_TO_BGP;
                    peer-as 64512;
                    description "Signaling VEPC VPN - 1001 - peering session (ext 1)";
                    family inet {
                        unicast;
                    }
                    local-address 11.10.1.3;
                    neighbor 11.10.1.1;
                    neighbor 11.10.1.2;
                }
            }
        }
    }
    sig_1002_vpn {
        instance-type virtual-router;
        description "Signaling VEPC VPN - 1002";
        interface ge-0/0/5.1002;
        interface lo0.1002;
        routing-options {
            autonomous-system 65001;
            static {
                route 1.0.0.1/32 {
                    next-hop 11.10.2.1;
                    metric 254;
                    bfd-liveness-detection {
                        description "BFD to vrouter primary IP."
                        minimum-interval 1000;
                        minimum-receive-interval 1000;
                        transmit-interval {
                            minimum-interval 1000;
                        }
                        multiplier 3;
                        session-mode multihop;
                     }
                }
            }
        }
        protocols {
            bgp {
                group sig_vsfo1ext1vlan1002 {
                    type external;
                    export REDISTRIBUTE_TO_BGP;
                    peer-as 64512;
                    description "Signaling VEPC VPN - 1002 - peering session (ext 1)";
                    family inet {
                        unicast;
                    }
                    local-address 11.10.2.3;
                    neighbor 11.10.2.1;
                    neighbor 11.10.2.2;
                }
            }
        }
    }
    sig_1003_vpn {
        instance-type virtual-router;
        description "Signaling VEPC VPN - 1003";
        interface ge-0/0/5.1003;
        interface lo0.1003;
        routing-options {
            autonomous-system 65001;
            static {
                route 1.0.0.1/32 {
                    next-hop 11.10.3.1;
                    metric 254;
                    bfd-liveness-detection {
                        description "BFD to vrouter primary IP."
                        minimum-interval 1000;
                        minimum-receive-interval 1000;
                        transmit-interval {
                            minimum-interval 1000;
                        }
                        multiplier 3;
                        session-mode multihop;
                     }
                }
            }
        }
        protocols {
            bgp {
                group sig_vsfo1ext1vlan1003 {
                    type external;
                    export REDISTRIBUTE_TO_BGP;
                    peer-as 64512;
                    description "Signaling VEPC VPN - 1003 - peering session (ext 1)";
                    family inet {
                        unicast;
                    }
                    local-address 11.10.3.3;
                    neighbor 11.10.3.1;
                    neighbor 11.10.3.2;
                }
            }
        }
    }
    radius_2000_vpn {
        instance-type virtual-router;
        description "Radius VEPC VPN - 2000";
        interface ge-0/0/5.2000;
        interface lo0.2000;
        routing-options {
            autonomous-system 65001;
            static {
                route 1.0.0.1/32 {
                    next-hop 11.20.0.1;
                    metric 254;
                    bfd-liveness-detection {
                        description "BFD to vrouter primary IP."
                        minimum-interval 1000;
                        minimum-receive-interval 1000;
                        transmit-interval {
                            minimum-interval 1000;
                        }
                        multiplier 3;
                        session-mode multihop;
                     }
                }
            }
        }
        protocols {
            bgp {
                group radius_vsfo1ext1vlan2000 {
                    type external;
                    export REDISTRIBUTE_TO_BGP;
                    peer-as 64512;
                    description "Radius VEPC VPN - 2000 - peering session (ext 1)";
                    family inet {
                        unicast;
                    }
                    local-address 11.20.0.3;
                    neighbor 11.20.0.1;
                    neighbor 11.20.0.2;
                }
            }
        }
    }
    radius_2001_vpn {
        instance-type virtual-router;
        description "Radius VEPC VPN - 2001";
        interface ge-0/0/5.2001;
        interface lo0.2001;
        routing-options {
            autonomous-system 65001;
            static {
                route 1.0.0.1/32 {
                    next-hop 11.20.1.1;
                    metric 254;
                    bfd-liveness-detection {
                        description "BFD to vrouter primary IP."
                        minimum-interval 1000;
                        minimum-receive-interval 1000;
                        transmit-interval {
                            minimum-interval 1000;
                        }
                        multiplier 3;
                        session-mode multihop;
                     }
                }
            }
        }
        protocols {
            bgp {
                group radius_vsfo1ext1vlan2001 {
                    type external;
                    export REDISTRIBUTE_TO_BGP;
                    peer-as 64512;
                    description "Radius VEPC VPN - 2001 - peering session (ext 1)";
                    family inet {
                        unicast;
                    }
                    local-address 11.20.1.3;
                    neighbor 11.20.1.1;
                    neighbor 11.20.1.2;
                }
            }
        }
    }
    radius_2002_vpn {
        instance-type virtual-router;
        description "Radius VEPC VPN - 2002";
        interface ge-0/0/5.2002;
        interface lo0.2002;
        routing-options {
            autonomous-system 65001;
            static {
                route 1.0.0.1/32 {
                    next-hop 11.20.2.1;
                    metric 254;
                    bfd-liveness-detection {
                        description "BFD to vrouter primary IP."
                        minimum-interval 1000;
                        minimum-receive-interval 1000;
                        transmit-interval {
                            minimum-interval 1000;
                        }
                        multiplier 3;
                        session-mode multihop;
                     }
                }
            }
        }
        protocols {
            bgp {
                group radius_vsfo1ext1vlan2002 {
                    type external;
                    export REDISTRIBUTE_TO_BGP;
                    peer-as 64512;
                    description "Radius VEPC VPN - 2002 - peering session (ext 1)";
                    family inet {
                        unicast;
                    }
                    local-address 11.20.2.3;
                    neighbor 11.20.2.1;
                    neighbor 11.20.2.2;
                }
            }
        }
    }
    radius_2003_vpn {
        instance-type virtual-router;
        description "Radius VEPC VPN - 2003";
        interface ge-0/0/5.2003;
        interface lo0.2003;
        routing-options {
            autonomous-system 65001;
            static {
                route 1.0.0.1/32 {
                    next-hop 11.20.3.1;
                    metric 254;
                    bfd-liveness-detection {
                        description "BFD to vrouter primary IP."
                        minimum-interval 1000;
                        minimum-receive-interval 1000;
                        transmit-interval {
                            minimum-interval 1000;
                        }
                        multiplier 3;
                        session-mode multihop;
                     }
                }
            }
        }
        protocols {
            bgp {
                group radius_vsfo1ext1vlan2003 {
                    type external;
                    export REDISTRIBUTE_TO_BGP;
                    peer-as 64512;
                    description "Radius VEPC VPN - 2003 - peering session (ext 1)";
                    family inet {
                        unicast;
                    }
                    local-address 11.20.3.3;
                    neighbor 11.20.3.1;
                    neighbor 11.20.3.2;
                }
            }
        }
    }
    radius_2004_vpn {
        instance-type virtual-router;
        description "Radius VEPC VPN - 2004";
        interface ge-0/0/5.2004;
        interface lo0.2004;
        routing-options {
            autonomous-system 65001;
            static {
                route 1.0.0.1/32 {
                    next-hop 11.20.4.1;
                    metric 254;
                    bfd-liveness-detection {
                        description "BFD to vrouter primary IP."
                        minimum-interval 1000;
                        minimum-receive-interval 1000;
                        transmit-interval {
                            minimum-interval 1000;
                        }
                        multiplier 3;
                        session-mode multihop;
                     }
                }
            }
        }
        protocols {
            bgp {
                group radius_vsfo1ext1vlan2004 {
                    type external;
                    export REDISTRIBUTE_TO_BGP;
                    peer-as 64512;
                    description "Radius VEPC VPN - 2004 - peering session (ext 1)";
                    family inet {
                        unicast;
                    }
                    local-address 11.20.4.3;
                    neighbor 11.20.4.1;
                    neighbor 11.20.4.2;
                }
            }
        }
    }
    radius_2005_vpn {
        instance-type virtual-router;
        description "Radius VEPC VPN - 2005";
        interface ge-0/0/5.2005;
        interface lo0.2005;
        routing-options {
            autonomous-system 65001;
            static {
                route 1.0.0.1/32 {
                    next-hop 11.20.5.1;
                    metric 254;
                    bfd-liveness-detection {
                        description "BFD to vrouter primary IP."
                        minimum-interval 1000;
                        minimum-receive-interval 1000;
                        transmit-interval {
                            minimum-interval 1000;
                        }
                        multiplier 3;
                        session-mode multihop;
                     }
                }
            }
        }
        protocols {
            bgp {
                group radius_vsfo1ext1vlan2005 {
                    type external;
                    export REDISTRIBUTE_TO_BGP;
                    peer-as 64512;
                    description "Radius VEPC VPN - 2005 - peering session (ext 1)";
                    family inet {
                        unicast;
                    }
                    local-address 11.20.5.3;
                    neighbor 11.20.5.1;
                    neighbor 11.20.5.2;
                }
            }
        }
    }
}

protocols {
    bfd {
        traceoptions {
            file bfd-trace;
            flag all;
        }
    }
    ospf {
        area 0.0.0.0 {
            interface ge-0/0/0.0 {
                priority 0;
                hello-interval 1;
                dead-interval 3;
            }
            interface lo0.0 {
                passive;
            }
        }
    }
}

routing-options {
    static {
        route 0.0.0.0/1 next-hop 192.168.25.1;
        route 128.0.0.0/1 next-hop 192.168.25.1;
    }
    router-id 172.31.2.1;
    autonomous-system 65001;
}


common/heat/env.py
from tcutils.util import get_random_cidr

src_cidr_svc = {
    "parameters":
    {
        # VN config
        "management_network": "management_vn",
        "left_vn": "left_vn",
        "right_vn": "right_vn",
        "left_vn_fqdn": "",
        "right_vn_fqdn": "",
        "network_policy_entries_policy_rule_src_addresses_subnet_ip_prefix": "10.10.10.3",
        #"network_policy_entries_policy_rule_src_addresses_subnet_ip_prefix": "10.10.10.3",
        "network_policy_entries_policy_rule_src_addresses_subnet_ip_prefix_len": "32",
        # VM config
        "flavor": "contrail_flavor_tiny",
        "image": "cirros",
        "left_vm1_name": "left_vm1",
        "left_vm2_name": "left_vm2",
        "right_vm_name": "right_vm",
        #ST Config
        "service_template_name": "test_cidr_template",
        "service_template_properties_version": 2,
        "service_template_properties_service_mode": "in-network",
        "service_template_properties_service_type": "firewall",
        "service_template_properties_interface_type_service_interface_type_1": "management",
        "service_template_properties_interface_type_service_interface_type_2": "left",
        "service_template_properties_interface_type_service_interface_type_3": "right",
        "service_template_properties_ordered_interfaces": True,
        "service_template_properties_service_virtualization_type": "virtual-machine",
        "domain": "default-domain",
        # SI Config
        "service_instance_name": "my-CIDR-SI",
        "svm1_name": "test_cidr_svm",
        "svm1_image": "tiny_in_net",
        "svm1_flavor": "contrail_flavor_tiny",
        "service_instance_fq_name": "",
        #PT Config
        "pt_name": "pt_test_cidr",
        #IPAM Config
        "network_ipam_refs_data_ipam_subnets_subnet_ip_prefix_1": '30.30.30.0',
        "network_ipam_refs_data_ipam_subnets_subnet_ip_prefix_len_1" : 24,
        "network_ipam_refs_data_ipam_subnets_subnet_ip_prefix_2" : '10.10.10.0',
        "network_ipam_refs_data_ipam_subnets_subnet_ip_prefix_len_2" : 24,
        "network_ipam_refs_data_ipam_subnets_subnet_ip_prefix_3" : '20.20.20.0',
        "network_ipam_refs_data_ipam_subnets_subnet_ip_prefix_len_3" : 24,
        "network_ipam_refs_data_ipam_subnets_addr_from_start_true" : "true",
        #Policy Config
        "policy_name": "cidr_policy",
        "policy_fq_name": "",
        "simple_action": "pass",
        "protocol": "any",
        "src_port_end": -1,
        "src_port_start": -1,
        "direction": "<>",
        "dst_port_end": -1,
        "dst_port_start": -1,
        "apply_services": "my-CIDR-SI",
    }
}

ecmp_pt = {
    "parameters":
    {"domain": "default-domain",
     "dst_port_end": -1,
     "protocol": "any",
     "service_template_properties_version": 2,
     "svm2_name": "pt_svm2",
     "image": "cirros",
     "dst_port_start": -1,
     "service_template_properties_service_type": "firewall",
     "service_template_properties_service_mode": "in-network-nat",
     "network_ipam_refs_data_ipam_subnets_subnet_ip_prefix_3":
     '%s' % get_random_cidr().split('/')[0],
     "network_ipam_refs_data_ipam_subnets_subnet_ip_prefix_2":
     "%s" % get_random_cidr().split('/')[0],
     "simple_action": "pass",
     "flavor": "contrail_flavor_tiny",
     "src_port_start": -1,
     "right_vn_fqdn": "",
     "service_template_properties_ordered_interfaces": "true",
     "left_vn": "left_vn",
     "network_ipam_refs_data_ipam_subnets_addr_from_start_true": "true",
     "left_vn_fqdn": "",
     "network_ipam_refs_data_ipam_subnets_subnet_ip_prefix_len_3": 24,
     "service_template_properties_interface_type_service_interface_type_3":
     "right",
     "network_ipam_refs_data_ipam_subnets_subnet_ip_prefix_len_1": 24,
     "service_template_properties_interface_type_service_interface_type_1":
     "management",
     "src_port_end": -1,
     "direction": "<>",
     "service_template_properties_flavor": "contrail_flavor_tiny",
     "service_template_properties_image_name": "tiny_nat_fw",
     "service_template_properties_interface_type_service_interface_type_2":
     "left",
     "service_instance_name": "pt_instance",
     "right_vn": "right_vn",
     "network_ipam_refs_data_ipam_subnets_subnet_ip_prefix_len_2": 24,
     "service_instance_fq_name": "",
     "service_template_name": "pt_template",
     "management_network": "management_vn",
     "left_vm_name": "left_vm",
     "right_vm_name": "right_vm",
     "service_instance_properties_ha_mode": "active-active",
     "network_ipam_refs_data_ipam_subnets_subnet_ip_prefix_1":
     "%s" % get_random_cidr().split('/')[0],
     "policy_name": "pt_policy",
     "svm1_name": "pt_svm1"}
}

pt_multi_inline = {
    "parameters":
    {"svc_vn_mgmt_prefix": "%s" % get_random_cidr().split('/')[0],
     "svc_vn_prefix_len": 24,
     "domain": "default-domain",
     "dst_port_end": -1,
     "protocol": "any",
     "service_template_properties_version": 2,
     "svm2_name": "pt_svm2",
     "svc_vn_right": "svc_vn_right",
     "service_template1_properties_image_name": "tiny_trans_fw",
     "service_instance1_name": "pt_instance1",
     "service_template_properties_service_type": "firewall",
     "service_template2_properties_service_mode": "in-network-nat",
     "network_ipam_refs_data_ipam_subnets_subnet_ip_prefix_3":
     "%s" % get_random_cidr().split('/')[0],
     "network_ipam_refs_data_ipam_subnets_subnet_ip_prefix_2":
     "%s" % get_random_cidr().split('/')[0],
     "simple_action": "pass",
     "flavor": "contrail_flavor_tiny",
     "src_port_start": -1,
     "service_template1_properties_service_mode": "transparent",
     "service_template1_name": "pt_template1",
     "right_vn_fqdn": "",
     "service_template_properties_ordered_interfaces": "true",
     "network_ipam_refs_data_ipam_subnets_addr_from_start_true": "true",
     "svc_vn_left_prefix": "%s" % get_random_cidr().split('/')[0],
     "network_ipam_refs_data_ipam_subnets_subnet_ip_prefix_len_3": 24,
     "service_template_properties_interface_type_service_interface_type_3":
     "right",
     "network_ipam_refs_data_ipam_subnets_subnet_ip_prefix_len_1": 24,
     "service_template_properties_interface_type_service_interface_type_1":
     "management",
     "image": "cirros",
     "src_port_end": -1,
     "service_instance2_fq_name": "",
     "service_template2_properties_image_name": "tiny_nat_fw",
     "service_template2_name": "pt_template2",
     "left_vn_fqdn": "",
     "service_template_properties_interface_type_service_interface_type_2":
     "left",
     "right_vn": "right_vn",
     "network_ipam_refs_data_ipam_subnets_subnet_ip_prefix_len_2": 24,
     "svc_vn_mgmt": "svc_vn_mgmt",
     "left_vm_name": "left_vm",
     "policy_name": "pt_policy",
     "dst_port_start": -1,
     "left_vn": "left_vn",
     "management_network": "management_vn",
     "service_instance1_fq_name": "",
     "right_vm_name": "right_vm",
     "service_template_properties_flavor": "contrail_flavor_tiny",
     "svc_vn_right_prefix": "%s" % get_random_cidr().split('/')[0],
     "service_instance2_name": "pt_instance2",
     "svc_vn_left": "svc_vn_left",
     "network_ipam_refs_data_ipam_subnets_subnet_ip_prefix_1":
     "%s" % get_random_cidr().split('/')[0],
     "direction": "<>",
     "svm1_name": "pt_svm1"}
}

vn = {
  u'parameters': {
    u'name': u'vn',
    u'ipam': u'default-domain:default-project:default-network-ipam',
    u'subnet': u'100.100.100.0', u'prefix': 24,
    u'transit': u'False'
  }
}

transit_net = {
  u'parameters': {
    u'name': u'transit-vn',
    u'ipam': u'default-domain:default-project:default-network-ipam',
    u'subnet': u'30.30.30.0', u'prefix': 24,
    u'transit': u'True'
  }
}

single_vm = {
  u'parameters': {
    u'vm_name': u'end-vm', u'net_id': u'',
    u'image': u'ubuntu-traffic', u'flavor': u'contrail_flavor_small'
  }
}

fip_pool = {
  u'parameters': {
    u'floating_pool': u'floating_pool', u'vn': u'',
  }
}

fip = {
  u'parameters': {
    u'project_name': u'', u'vmi': u'',
    u'floating_pool': u'',
  }
}

intf_rt_table = {
  u'parameters': {
    u'intf_rt_table_name': u'intf_rt_tbl', u'route_prefix': u'',
    u'si_fqdn': u'', u'si_intf_type': u'',
  }
}

vms = {
  u'parameters': {
    u'right_vm_name': u'right-vm', u'right_net_id': u'',
    u'left_vm_name': u'left-vm', u'left_net_id': u'',
    u'image': u'ubuntu-traffic', u'flavor': u'',
    u'availability_zone': u'',
  }
}

svc_tmpl = {
  u'parameters': {
    u'domain':u'',
    u'name': u'st1',
    u'service_interface_type_list': u'management,left,right',
    u'image': u'',
    u'static_routes_list': u'False,False,False',
    u'mode': u'',
    u'flavor': u'',
    u'service_scaling': 'False',
    u'type': u'firewall',
    u'shared_ip_list': u'False,False,False'
  }
}

svc_tmpl_v2 = {
  u'parameters': {
    u'domain':u'',
    u'name': u'',
    u'image': u'', u'flavor': u'',
    u'mode': u'', u'type': u'firewall',
    u'service_scaling': u'False',
    u'left_shared': u'False', u'left_static': u'False',
    u'right_static': u'False', u'right_shared': u'False'
  }
}

svc_tmpl_pt_v2 = {
  u'parameters': {
    u'domain':u'',
    u'name': u'',
    u'mode': u'', u'type': u'firewall',
  }
}

svc_inst = {
  u'parameters': {
    u'service_template_fq_name': u'default-domain:st1',
    u'left_net_id': u'3f162ab2-85ff-4ad0-8161-9ae4633e7359',
    u'service_instance_name': u'si1',
    u'max_instances': u'1',
    u'right_net_id': u'e6f9e85b-5816-4818-bea6-089262c63f5d'
  }
}

svc_inst_v2 = {
  u'parameters': {
    u'service_template_fq_name': u'', u'service_instance_name': u'',
    u'max_instances': 1, u'ha': u'active-active',
    u'left_net_id': u'', u'right_net_id': u''
  }
}

svc_inst_dual_v2 = svc_inst_v2

svc_inst_pt_v2 = {
  u'parameters': {
    u'image': u'', u'flavor': u'', u'svm_name': 'svm',
    u'service_template_fq_name': u'', u'service_instance_name': u'',
    u'mgmt_net_id': u'',u'left_net_id': u'', u'right_net_id': u'',
    u'security_group_ref':u'',
    u'availability_zone': u'',
  }
}

svc_inst_pt_dual_v2 = svc_inst_pt_v2

pt_svm = {
  u'parameters': {
    u'image': u'tiny_nat_fw', u'flavor': u'contrail_flavor_tiny', u'svm_name': u'',
    u'si_fqdn': u'',u'mgmt_net_id': u'', u'left_net_id': u'', u'right_net_id': u'',u'intf_rt_table_fqdn': u'',
    u'def_sg_id': u''
  }
}

pt_si = {
  u'parameters': {
    u'service_template_fq_name': u'', u'service_instance_name': u'',
    u'left_net_id': u'', u'right_net_id': u'', u'mgmt_net_id': u''

  }
}

svc_chain = {
  u'parameters': {
    u'policy_name': u'',
    u'src_vn_id': u'',
    u'dst_vn_id': u'',
  }
}

svc_chain_v2 = {
  u'parameters': {
    u'policy_name': u'',
  }
}
